Smt/Prop.lean

```diff
@@ -35,12 +34,43 @@ open Translator Term
       return existsT n.toString tmT tmB
   | _ => return none
 
-@[smtTranslator] def replaceDecIte : Translator
-  | app (app (app (const `ite _) _) e) _ => do
-    let tmE ← applyTranslators! e
-    return appT (symbolT "ite") tmE
+/- @Eq.rec : {α : Sort u_2} →
+  {a : α} → {motive : (a_1 : α) → a = a_1 → Sort u_1} → motive a (_ : a = a) → {a_1 : α} → (t : a = a_1) → motive a_1 t -/
+@[smtTranslator] def replaceEqRec : Translator
+  | app (app (app (app (app (app (const `Eq.rec _) _) _) _) e) _) _ => do
+    trace[smt.debug.replaceEqRec] "found eq_rec body : {e}"
+    applyTranslators? e
   | _ => return none
 
+def emitIte (cond : Expr) (_inst : Expr) (t : Unit → TranslationM Term) (f : Unit → TranslationM Term)
+    : TranslationM (Option Term) := do
+  return mkApp3 (symbolT "ite") (← applyTranslators! cond) (← t ()) (← f ())
```

Comment:
What is `_inst` for? Also, why is the type of `t` and `f` `Unit → TranslationM Term` and not `TranslationM Term`?

Reply:
The `_inst` was a leftover from an attempt at constant branch elimination, removed. `Unit` was just my mistake, I forgot that `TranslationM Term` is already a lazy term.